.github/workflows/python-package.yml

```diff
@@ -1,47 +1,48 @@
+# https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
 name: Python package
 
 on:
   push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'            # release
+      - '[0-9]+.[0-9]+.[0-9]+a[0-9]+'     # alpha
+      - '[0-9]+.[0-9]+.[0-9]+b[0-9]+'     # beta
+      - '[0-9]+.[0-9]+.[0-9]+rc[0-9]+'    # release candidate
+      - '[0-9]+.[0-9]+.[0-9]+.dev[0-9]+'  # dev (not semver compliant)
+      - '[0-9]+.[0-9]+.[0-9]+.post[0-9]+' # post (not semver compliant)
   pull_request:
 
 jobs:
   cqa:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
+
+    - uses: awalsh128/cache-apt-pkgs-action@latest
+      with:
+        packages: zsh
+        version: 1.0
 
     - name: Set up Python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.10"
+        python-version: "3.12"
         cache: pip
-        cache-dependency-path: '**/setup.cfg'
+        cache-dependency-path: '**/pyproject.yaml'
 
     - name: Install test dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install --use-deprecated=legacy-resolver -e .[dev]
+        pip install --use-deprecated=legacy-resolver -e .[dev,test]
```

Comment:
It might be good to consider breaking out test vs dev deps in the example repo

Reply:
draft PR: https://github.com/biocommons/biocommons.example/pull/53